extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.js

```diff
@@ -0,0 +1,187 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActionTrust} from '../../../src/action-constants';
+import {Services} from '../../../src/services';
+import {dev, user} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {tryParseJson} from '../../../src/json';
+
+
+/** @const {string} */
+const TAG = 'amp-viewer-assistance';
+
+/** @const {string} */
+const GSI_TOKEN_PROVIDER = 'actions-on-google-gsi';
+
+export class AmpViewerAssistance {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    const assistanceElement = ampdoc.getElementById('amp-viewer-assistance');
+
+    /** @private {boolean} */
+    this.enabled_ = !!assistanceElement;
+    if (!this.enabled_) {
+      return;
+    }
+
+    /** @const @private */
+    this.ampdoc_ = ampdoc;
+
+    /** @const @private {!Element} */
+    this.assistanceElement_ = dev().assertElement(assistanceElement);
+
+    /** @const @private {JsonObject|null|undefined} */
+    this.configJson_ = tryParseJson(this.assistanceElement_.textContent, e => {
+      throw user().createError(
+          'Failed to parse "amp-viewer-assistance" JSON: ' + e);
+    });
+
+    /** @private @const {!../../../src/service/viewer-impl.Viewer} */
+    this.viewer_ = Services.viewerForDoc(ampdoc);
+
+    /** @private @const {!../../../src/service/action-impl.ActionService} */
+    this.action_ = Services.actionServiceForDoc(this.assistanceElement_);
+
+    /** @private @const {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(ampdoc.win);
+  }
+
+  /**
+   * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   * @return {?Promise}
+   * @private
+   */
+  actionHandler_(invocation) {
+    const {method, args} = invocation;
+    if (method == 'updateActionState' && !!args) {
+      this.viewer_.sendMessageAwaitResponse(method, args).catch(error => {
+        user().error(TAG, error.toString());
+      });
+    } else if (method == 'signIn') {
+      this.requestSignIn_();
+    }
+
+    return null;
+  }
+
+  /**
+   * @private
+   * @restricted
+   */
+  start_() {
+    if (!this.enabled_) {
+      user().info(
+          TAG, 'Invalid AMP Action - no "id=amp-viewer-assistance" element');
+      return this;
+    }
+    return this.viewer_.isTrustedViewer().then(isTrustedViewer => {
+      if (!isTrustedViewer &&
+         !isExperimentOn(this.ampdoc_.win, 'amp-viewer-assistance-untrusted')) {
```

Comment:
Is this intended to be enabled eventually?

Reply:
No, the experiment was our method of forgoing the trusted viewer check with testing locally since local viewers are inherently not trusted.

If there are other ways of accomplishing the same effect, we are open to suggestions.

Reply:
Yes, this is specifically for developing/testing with a local viewer using a remote (or local) runtime. Does that make sense?